mm2src/coins_activation/src/l2/init_l2.rs

```diff
@@ -39,7 +39,13 @@ pub trait InitL2ActivationOps: Into<MmCoinEnum> + Send + Sync + 'static {
     type ValidatedParams: Clone + Send + Sync;
     type CoinConf: Clone + Send + Sync;
     type ActivationResult: serde::Serialize + Clone + Send + Sync;
-    type ActivationError: From<RegisterCoinError> + NotEqual + SerMmErrorType + Clone + Send + Sync;
+    type ActivationError: From<CoinIsAlreadyActivatedErr>
```

Comment:
I believe `CoinIsAlreadyActivatedErr` should be removed all together from the code and `RegisterCoinError` to be used instead of it since they serve the same purpose. You will be using `RegisterCoinError` in `add_token` and `add_l2` functions, and you should implement `From<RegisterCoinError> for EnableTokenError`.